### PR TITLE
Direct Generators support

### PR DESCRIPTION
**Pull request by bermanjosh**
This PR adds the ability to modify the `Direct Generators` object within the `Phrase Suggester`. 

[Docs link](https://www.elastic.co/guide/en/elasticsearch/reference/5.6/search-suggesters-phrase.html#_direct_generators)

### CODE REVIEW

**Comment by bermanjosh** [comment]
@MichaelXavier @bitemyapp - just bumping this - thanks.

File: src/Database/V5/Bloodhound/Types.hs [hunk 1]
```diff
@@ -5293,6 +5296,7 @@ data PhraseSuggester =
                   , phraseSuggesterShardSize :: Maybe Int
                   , phraseSuggesterHighlight :: Maybe PhraseSuggesterHighlighter
                   , phraseSuggesterCollate :: Maybe PhraseSuggesterCollate
+                  , phraseSuggesterCandidateGenerators :: [DirectGenerators]
```
Comment (MichaelXavier) [inline src/Database/V5/Bloodhound/Types.hs]: Wouldn't the addition of this field to an exported type constitute a B version number change instead of a C?

**Comment by MichaelXavier** [comment]
Had one minor comment about it possibly being a breaking change since it technically adds a field to a fully exported constructor. Also curious if this should be added to the V1 and V2 modules tree as well, a cursory look seems like [direct generators were around in 1.x](https://www.elastic.co/guide/en/elasticsearch/reference/1.7/search-suggesters-phrase.html#_direct_generators).

Beyond that, this falls into my normal evaluation of an ES feature addition: I don't really know what this does but as long as it works and doesn't block future feature additions, seems good to me.

**Comment by bermanjosh** [comment]
@MichaelXavier I ported the changes to V1 and bumped the version to 0.16. 

All the old tests are passing, but 8.2 isn't. That's a separate problem, I don't think this PR is the right place to deal with it. Let me know if you disagree.

**Comment by bitemyapp** [comment]
@bermanjosh @MichaelXavier  wait on the merge until I can unbreak master. Should've put that in a topic branch. Thought a GHC upgrade wouldn't be a big deal. This is why we don't make exceptions to the rules eh?

**Comment by bitemyapp** [comment]
Pull from master, it's clean now.

**Comment by bermanjosh** [comment]
@bitemyapp Done & working - thanks

**Comment by bermanjosh** [comment]
@bitemyapp Bump - thanks.

**Comment by bitemyapp** [comment]
@bermanjosh Thank you!

**Comment by bermanjosh** [comment]
Thanks to you!

**Pull request merged**